repo2docker/buildpacks/julia/__init__.py

```diff
@@ -132,20 +158,53 @@ def get_assemble_scripts(self):
         any needed Python packages found in environment.yml.
 
         """
-        require = self.binder_path('REQUIRE')
-        return super().get_assemble_scripts() + [(
-            "${NB_USER}",
-            # Install and pre-compile all libraries if they've opted into it.
-            # In v0.6, Pkg.resolve() installs all the packages, but in v0.7+, we
-            # have to manually Pkg.add() each of them (since the REQUIRES file
-            # format is deprecated).
-            # The precompliation is done via `using {libraryname}`.
-            r"""
-            julia /tmp/install-repo-dependencies.jl "%(require)s"
-            """ % {"require": require}
-            # TODO: For some reason, `rm`ing the file fails with permission denied.
-            # && rm /tmp/install-repo-dependencies.jl
-        )]
+        if self.julia_env_exists:
+            return super().get_assemble_scripts() + [
+                (
+                    "${NB_USER}",
+                    # HACK: Can't seem to tell IJulia to install in sys-prefix
+                    # FIXME: Find way to get it to install under /srv and not $HOME?
+                    r"""
+                    julia -e 'using Pkg; Pkg.add("IJulia"); using IJulia; installkernel("Julia", "--project={0}");'
+                    """.format('~')
```

Comment:
@betatim & @NHDaly This is a place where I'm stuck. What we need to do here is pass the absolute path to the root of the project folder inside the docker image (instead of just `~`). Is there a way that I can get that path at this point? For example, when I run some test repo on the public mybinder.org website, it seems I would have to pass `/home/jovyan` here. When I run this on my mac, it seems to use my own username instead.

Reply:
I think `REPO_DIR` is the variable to look at for that.

Reply:
Yep, that was it, thanks!